### DISCUSSION TITLE
copy-on-select should trigger copy only on mouse release

### DISCUSSION THREAD

**Discussion by applebait** [ghostty-org/ghostty#4121] [Feature Requests, Ideas] [answered]
I think the title says it all.
If I'm selecting a big line, somewhere before ending the selection I already have a copy on the clipboard, then I finish it, and end up with 2 entries in the clipboard, where the first one is trash…
I think this feature would make more sense if on mouse button release.

Thanks

**Comment by JakeWharton** [comment]
You already created a discussion here: https://github.com/ghostty-org/ghostty/discussions/3874

This seems like an exact duplicate.

**Reply by Ch00k** [reply]
Since #3874 is closed, could this be un-marked as answer?

**Reply by JakeWharton** [reply]
I didn't have the ability to do that. However, someone just filed https://github.com/ghostty-org/ghostty/discussions/4750 so let's take the discussion there.

**Comment by Ch00k** [comment]
#3874 is closed, which I noticed only when I commented on it.

In my case a new clipboard entry is created for (almost) each character in a line when I drag the selection through this line.

The screenshot shows the clipboard contents through [Alfred](https://www.alfredapp.com/).

<img width="1425" alt="Screenshot 2025-01-03 at 10 05 29" src="https://github.com/user-attachments/assets/df55c44f-405d-4528-b570-e549cc31e5dd" />

**Comment by cgjosephlee** [comment]
Same issue here.

**Comment by pluiedev** [comment, marked as answer]
Superseded by #4800
